include/swift/AST/DiagnosticsSema.def

```diff
@@ -4452,6 +4452,9 @@ ERROR(property_wrapper_type_requirement_not_accessible,none,
       "more restrictive access than its enclosing property wrapper type %3 "
       "(which is %select{private|fileprivate|internal|public|open}4)",
       (AccessLevel, DescriptiveDeclKind, DeclName, Type, AccessLevel))
+ERROR(property_wrapper_ambiguous_enclosing_self_subscript, none,
+      "property wrapper type %0 has multiple enclosing-self subscripts %1",
+      (Type, DeclName))
```

Comment:
Does it work if you have something like this, to handle a value-semantics wrapper type?

```
// Read only access from a read only base
subscript(_enclosingInstance: EnclosingSelf, storage: KeyPath<EnclosingSelf, Self>) -> Wrapped { get }

// Writable projection from a writable base
subscript(_enclosingInstance: inout EnclosingSelf, storage: WritableKeyPath<EnclosingSelf, Self>) -> Wrapped { get set }
```